src/redux/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,68 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = {
+  rocketsList: [],
+  loading: true,
+};
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  try {
+    const resp = await fetch(url);
+    if (!resp.ok) {
+      throw new Error('Something went wrong');
+    }
+    const data = await resp.json();
+    return data;
+  } catch (error) {
+    throw new Error(error);
+  }
+});
```

Comment:
- [ ] Kindly filter the data from the rockets API as well using the requirement.

  ![image](https://github.com/otmaneechchafyky/space-travelers-hub/assets/36542986/98aef158-d37c-46a6-a431-afc3c8a03720)